projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-jetpack-token-subscription-service.php

```diff
@@ -32,6 +33,10 @@ public static function available() {
 	 * @return int The site ID.
 	 */
 	public function get_site_id() {
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			return get_current_blog_id();
+		}
+
 		return \Jetpack_Options::get_option( 'id' );
```

Comment:
this works in WPCOM and Jetpack, I think there's no need for the previous `if`

Reply:
I'll try this. Previously, I had just copied this logic over from `WPCOM_Token_Subscription_Service` which uses `get_current_blog_id`. Ultimately this gets overridden by `WPCOM_Online_Subscription_Service` anyway, which uses `get_current_blog_id`.

Reply:
> Ultimately this gets overridden by WPCOM_Online_Subscription_Service anyway, which uses get_current_blog_id.

I would definitely remove it as it will add confusion for future devs